app/uk/gov/hmrc/teamsandrepositories/connectors/GithubConnector.scala

```diff
@@ -115,6 +120,61 @@ class GithubConnector @Inject()(
     )
   }
 
+  /**
+   * A combinator to augment a list of repositories with the branch protection policy of their default branches.
+   *
+   * This is performed after-the-fact and in-memory to avoid making an additional call-per-repository to the GitHub API
+   * (which would slow down the scheduled job and risk an altercation with GitHub's rate limiter.)
+   */
+  def withBranchProtectionPolicies(action: => Future[List[GhRepository]]): Future[List[GhRepository]] = {
```

Comment:
I think augmenting the repositories with branch protection information at this level (in `GithubConnector`, as an after-the-fact combinator) results in a smaller change footprint. It also sidesteps threading it through some fairly non-trivial logic in `PersistingService` if it were to live in `PersistingService.persistTeamRepoMapping`.

_BUT_, it comes at the expense of an additional call for each GitHub team.

What we could do, to get the best of both worlds, is reimplement `GithubConnector.getReposForTeam()` and `GithubConnector.getRepos()` in terms of the GraphQL API, such that no additional requests are required at all.